output/transform/merge_partition_sizes.go

```diff
@@ -0,0 +1,56 @@
+package transform
+
+import (
+	snapshot "github.com/pganalyze/collector/output/pganalyze_collector"
+	"github.com/pganalyze/collector/state"
+)
+
+func mergePartitionSizes(s snapshot.FullSnapshot, newState state.PersistedState, ts state.TransientState, databaseOidToIdx OidToIdx) snapshot.FullSnapshot {
+	for idx, info := range s.IndexInformations {
+		rel := s.RelationInformations[info.RelationIdx]
+		if !rel.HasParentRelation || rel.PartitionBoundary == "" {
+			continue
+		}
+		for parentIdx, pi := range s.IndexInformations {
```

Comment:
It isn't ideal to use a nested loop like this, but because the data is so spread out across different data structures I found it difficult to implement this at a step before the snapshot is built.

It also implicitly relies on the info and statistics arrays being a 1-1 mapping. In `output/transform/postgres_relations.go` I've already made it so `IndexStatistics` exist for all `IndexInformations`, but in preparing this PR I ran into an error because that hasn't been applied to `RelationInformations` and `RelationStatistics` yet. There's the open question: why do we skip creating statistics if the diff is missing? What's the harm in sending zeroes?

```
Panic: runtime error: index out of range [899] with length 899
goroutine 1497 [running]:
runtime/debug.Stack()
	/opt/homebrew/Cellar/go/1.23.2/libexec/src/runtime/debug/stack.go:26 +0x64
github.com/pganalyze/collector/runner.capturePanic.func1()
	/Users/sean/code/pganalyze/collector/runner/full.go:78 +0x64
panic({0x1038e0fe0?, 0x14000870000?})
	/opt/homebrew/Cellar/go/1.23.2/libexec/src/runtime/panic.go:785 +0x124
github.com/pganalyze/collector/output/transform.mergePartitionSizes({{{}, {}, {}, 0x0}, 0x0, {0x0, 0x0, 0x0}, 0x0, 0x0, ...}, ...)
	/Users/sean/code/pganalyze/collector/output/transform/merge_partition_sizes.go:29 +0x1e0